h2o-automl/src/main/java/water/rapids/ast/prims/mungers/AstTargetEncoderTransform.java

```diff
@@ -35,24 +37,23 @@ public int nargs() {
   @Override
   public ValFrame apply(Env env, Env.StackHelp stk, AstRoot asts[]) {
 
-    String[] encodingMapKeys = getEncodingMapKeys(asts);
+    String[] encodingMapKeys = getEncodingMapKeys(env, stk, asts);
     Frame[] encodingMapFrames = getEncodingMapFrames(env, stk, asts);
     Frame frame = getFrameToTransform(env, stk, asts);
-    String[] teColumnsToEncode = getTEColumns(asts);
+    String[] teColumnsToEncode = getTEColumns(env, stk, asts);
     byte dataLeakageHandlingStrategy = getDataLeakageHandlingStrategy(env, stk, asts);
     String targetColumnName = getTargetColumnName(env, stk, asts);
     String foldColumnName = getFoldColumnName(env, stk, asts);
     boolean withBlending = getWithBlending(env, stk, asts);
-    double inflectionPoint = getInflectionPoint(env, stk, asts);
-    double smoothing = getSmoothing(env, stk, asts);
+
+    BlendingParams params = getBlendingParams(env, stk, asts, withBlending);
```

Comment:
maybe replace with

    BlendingParams params = withBlending ? getBlendingParams(env, stk, asts) : null;

?

Reply:
yeap, thank you!